src/helpers/generate-matrix.ts

```diff
@@ -17,11 +19,42 @@ import { chunk } from 'lodash';
 export class GenerateMatrix extends HelperInputs {
   paths = '';
   batches?: string;
+  load_balancing_sizes?: string;
 }
 
-export const generateMatrix = ({ paths, batches = '1' }: GenerateMatrix) => {
+export const generateMatrix = ({ paths, batches: _batches = '1', load_balancing_sizes }: GenerateMatrix) => {
   const matrixValues = paths.split(/[\n,]/);
-  return {
-    include: chunk(matrixValues, Math.ceil(matrixValues.length / Number(batches))).map(chunk => ({ path: chunk.join(',') }))
-  };
+  const batches = Number(_batches);
+  if (!load_balancing_sizes) {
+    return {
+      include: chunk(matrixValues, Math.ceil(matrixValues.length / batches)).map(chunk => ({ path: chunk.join(',') }))
+    };
+  } else {
+    const loadBalancingSizes = load_balancing_sizes.split(/[\n,]/).map(size => Number(size));
+    if (loadBalancingSizes.length !== matrixValues.length) throw new Error('load_balancing_sizes have the same length as paths');
+    const targetLoadSize = loadBalancingSizes.reduce((acc, size) => acc + size, 0) / batches;
+    const loadBalancedPaths: string[] = [];
+    let currentLoadSize = 0;
+    let currentBatch: string[] = [];
+    matrixValues.forEach((path, index) => {
+      const possibleLoadSize = currentLoadSize + loadBalancingSizes[index];
+      if (Math.abs(possibleLoadSize - targetLoadSize) <= Math.abs(loadBalancingSizes[index] - targetLoadSize)) {
```

Comment:
this is meant to push following loads into the next batch in the case that they result in the current load far exceeding the `targetLoadSize` (for example, without this, with a `targetLoadSize` of `5` and a `currentLoadSize` of `4`, a new load size of `20` would be lumped into the current batch. we obviously would like to avoid that, since such an outlier should be in its own batch)

i can put a comment in the code if that would be helpful for maintainability